spectral: unstable 2019-08-30 -> 817, olm: cmake+clean

dtzWill opened the pull request:
###### Motivation for this change

Update!

If crashes on launch, and you've previously used spectral,
I've found sometimes nuking paths like
`~/.cache/ENCOM/Spectral/qmlcache` or even
`~/.cache/ENCOM/Spectral` seems to fix things.

Not sure what is going on that would cause that, just a heads-up.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @dtzWill (me :innocent:)

dtzWill committed:
spectral: unstable-2019-08-30 -> 817

* libsecret, now req'd
* url updates

dtzWill committed:
olm: build w/cmake, simplify expression (somewhat optimistically)

dtzWill commented:
Oh, olm needs to be built w/cmake, missed that when picking this over.  Pulling that in, let's see how that looks...

dtzWill commented:
(using github's reviewer suggestions, thank you!)

colemickens commented on pkgs/applications/networking/instant-messengers/spectral/default.nix:
```diff
@@ -14,19 +14,17 @@ let qtkeychain-qt5 = qtkeychain.override {
     url = "https://gitlab.com/b0/spectral.git";
-    rev = "23237d0cb27a0b15c8e81c0e2311cc897302e6ba";
-    sha256 = "11jfq97fa90r8wrahb6aasvmi7k7hr61lqgnbknsr9akn4wgvrh8";
+    rev = "refs/tags/${version}";
```
```
fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
Unable to checkout refs/tags/ref/tags/817 from https://gitlab.com/spectral-im/spectral.git.
```

colemickens reviewed (changes requested):
Might be nice to have this squashed a bit more, also.

And thanks! I've been trying my hand at this a few times and had concluded olm needed help that I wasn't prepared to give it. :)

dtzWill commented:
On Sun, 12 Jan 2020 17:03:39 -0800, Cole Mickens <notifications@github.com> wrote:
> colemickens requested changes on this pull request.
> 
> Might be nice to have this squashed a bit more, also.

Done!

> 
> And thanks! I've been trying my hand at this a few times and had concluded olm needed help that I wasn't prepared to give it. :)
> 
> >    pname = "spectral";
> -  version = "unstable-2019-08-30";
> +  version = "817";
>  
>    src = fetchgit {
>      url = "https://gitlab.com/b0/spectral.git";
> 
> update url, org is now spectral-im

Included, thanks!

Updated homepage too, https://spectral.im .

> 
> >  
>    src = fetchgit {
>      url = "https://gitlab.com/b0/spectral.git";
> -    rev = "23237d0cb27a0b15c8e81c0e2311cc897302e6ba";
> -    sha256 = "11jfq97fa90r8wrahb6aasvmi7k7hr61lqgnbknsr9akn4wgvrh8";
> +    rev = "refs/tags/${version}";
> 
> ```
> fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
> Unable to checkout refs/tags/ref/tags/817 from https://gitlab.com/spectral-im/spectral.git.
> ```

Strange I didn't see this, dunno.

Switched to `rev=version;` hopefully that fixes things.

> 
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/NixOS/nixpkgs/pull/76767#pullrequestreview-341613426Non-text part: text/html

colemickens commented:
The rev/version/ref-tag  was me applying the patch wrong. I'd left out the `rec` so version was empty. Turns out your "fix" was unnecessary but would've fixed my issue. Problem is, I noticed before sending the review but forgot to fix my comment, sorry about that.

Anyway, it looks good to me and is a VERY welcome upgrade, I'm running it now, no issues with upgrading or havign to delete data. Thanks! :)

colemickens commented:
This fixes #77034 btw.

colemickens reviewed (approved):
LGTM!

dtzWill commented:
Great, thanks!
